pkg/services/pluginsintegration/serviceregistration/serviceregistration.go

```diff
@@ -58,35 +58,16 @@ func (s *Service) RegisterExternalService(ctx context.Context, pluginID string,
 
 		enabled = (settings != nil) && settings.Enabled
 	}
-
-	impersonation := extsvcauth.ImpersonationCfg{}
-	if svc.Impersonation != nil {
-		impersonation.Permissions = toAccessControlPermissions(svc.Impersonation.Permissions)
-		impersonation.Enabled = enabled
-		if svc.Impersonation.Groups != nil {
-			impersonation.Groups = *svc.Impersonation.Groups
-		} else {
-			impersonation.Groups = true
-		}
-	}
-
 	self := extsvcauth.SelfCfg{}
 	self.Enabled = enabled
```

Comment:
I guess we can clean up this `.Enabled` too since it's going to always be enabled?

Reply:
If I'm not mistaken, for app plugins it can be disabled:
```
enabled = (settings != nil) && settings.Enabled
```